### PR TITLE
Features

### LINKED ISSUE #687: Dynamic Group - Arcs
An "Arc" option for a Dynamic group. Where one can define the angle of the arc and length.

Currently using clockwise/counterclockwise and radius, but these assume a full circle. In order to clump elements along an edge of the "circle", blank placeholders need to be made to fill the circle and force the visible objects in place.

### LINKED ISSUE #1222: Preserve Option Name and Option Key when changing Custom Option Type
**Is your feature request related to a problem? Please describe.**
At the moment when changing the type of custom option removes the variable and name given to it whereas the description remains the same. 
**Describe the solution you'd like**
I'd like to see a change that keeps the variable assigned to the custom option while keeping it's name too.

### PR DESCRIPTION
**Pull request by emptyrivers**
# Description

Add limit option to dynamic group (implements [857](https://www.wowace.com/projects/weakauras-2/issues/857))
Add Arc length option to dynamic group (closes #687)
Preserve option name/key when changing custom option type (closes #1222)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Smoke test

### CODE REVIEW

**Comment by emptyrivers** [comment]
I also added a change that closes #1184.

File: WeakAuras/RegionTypes/DynamicGroup.lua [hunk 1]
```diff
@@ -351,6 +354,7 @@ local growers = {
     local radius = data.radius or 0
     local limit = data.useLimit and data.limit or math.huge
     local sAngle = (data.rotation or 0) * math.pi/180
+    local arc = (data.arcLength or 0) * math.pi/180
```
Comment (Stanzilla) [inline WeakAuras/RegionTypes/DynamicGroup.lua]: nit: spaces, also above

File: WeakAuras/RegionTypes/DynamicGroup.lua [hunk 2]
```diff
@@ -381,6 +385,7 @@ local growers = {
...
```
Comment (Stanzilla) [inline WeakAuras/RegionTypes/DynamicGroup.lua]: nit: spaces, also above

File: WeakAurasOptions/RegionOptions/DynamicGroup.lua [hunk 3]
```diff
@@ -281,11 +273,55 @@ local function createOptions(id, data)
         WeakAuras.ResetMoverSizer();
       end
     },
-    spacer = {
+    -- border/background options
+    -- TODO: use Weakauras.BorderOptions for these instead
```
Comment (Stanzilla) [inline WeakAurasOptions/RegionOptions/DynamicGroup.lua]: is that something you want to do this PR or in the future?

File: WeakAurasOptions/RegionOptions/DynamicGroup.lua [hunk 4]
```diff
@@ -281,11 +273,55 @@ local function createOptions(id, data)
...
+    },
+    background = {
+      type = "select",
+      width = WeakAuras.normalWidth,
+      dialogControl = "LSM30_Background",
+      name = L["Background"],
+      order = 20,
+      values = function()
+        local list = {};
+        for i,v in pairs(AceGUIWidgetLSMlists.background) do
```
Comment (Stanzilla) [inline WeakAurasOptions/RegionOptions/DynamicGroup.lua]: nit: spaces

**Comment by Stanzilla** [comment]
Do you maybe want to prefix your commits with `DynamicGroups:` or something like that? would probably look nicer in the history and something we should maybe do in general?

File: WeakAuras/RegionTypes/DynamicGroup.lua (hunk 2 above)
Comment (emptyrivers) [reply WeakAuras/RegionTypes/DynamicGroup.lua]: ? what spaces

Comment (Stanzilla) [reply WeakAuras/RegionTypes/DynamicGroup.lua]:
```suggestion
    local arc = (data.arcLength or 0) * math.pi / 180
```

**Comment by mrbuds** [comment]
Group missing border on empty icon (i dont know what steps to reproduce, using the aura bellow)
prev

limit bug (the icon doesn't have a cooldown, so i think it's showing another spell but didnt update icon
prev

```
!nBvWYjooq0VM5WEyRcZqYDcXmdvXyyroZU7fLiSegTrwYLKCiWb(23wTSnegYmzVTvLYrUTuRx)6x3n0eAoL4OK7fVqjvucp8hyQ0A2rjFpDv(SjJNtjRnwUWEhR4zU1utj3PKhoWS8JpTKzl2wj0Ek51fB24eEA2akz)zRz6ITg7sJmSPjPz5PR684eJYyHlmljSZSH4ZpJphfmchaqcPMvicVswdiaWwJMFog(cZALoJ(4t31)9JpneU5cV0ODyq6zwpSa8OulBxTbw62IRH38wzzPW6I4jV3qyjxSUzZM891ck5RPZxo9bGv84RSgldUlxTqPMXDrp7AwlEbyfcCk5RuYJtgtYFKKpEf4SMaai1k2EW59BDPvGBLSmDo4C0gCzcMYVLs0SkHRdPn6tqZhnvyunv6RXuZ0NseYkwGroNqAJcTrdp5qSe2WJrJorHrdHe4)kMu)X3DpX(X2pCcMswQpjpSgFePbCRmmuv24epwOyohDnY3fbJvnkVSJyIF9sRo5bXLg9rkkiN7eH(ogsG(VGPAPsuQsjjGAx(Qq1zjLxko3QvucioQr47HeMSiKbQ72)m9uRbYP3brKScKaz3eib1MlknCXK4WGs1lSAM67GQejJeaoodK3abPRqO5sDzi3RHSP2pf07qKqwn((zpqGOXwmxOlb5t2NVDqhm6QmhEcx47JqcEoImaIsOeltSl0lqY)tjh9cKzTGqeQmyEgsoXI7PbRXiNmzvAAwOSWdFyrDxfiQr1BKLX1nb)V52BUHn6FoSNmlEprx8qFhhsiZLDlEuU8mxzzCzJdcIbdqPmwmaALWT0x)wQGUyqQ0lE1NmfO(O7sgg8NrXn705WNs1S1kbpe0)Y2wDNJStciforrJZBQcU5HAEq6qAA)F3wrzcGcjedHtaKCayrvxl6ggqhLm1kpC8P)OHXd87XNYHG4zHOECqS7xfQiqP5pQzUcB78YIN3FFBnx4CNU2yh3eSJBc2Xn5uhxKUA9)DlYZx8TvZ(YxZpdPtvSsyQXIhYNpllLs2kKLBdsObOWK5dxkg0DLUVtH51QHdv5AO7hsoVtP7)N7Kv8Xy4HtaIeGsC25SmYS7tJPOyrW6gV3Oxa8amNOnP8bsEyD1FT7V)6VDt1KLVW)5J1CBbfHccjxqz(25z)SjyxmC6DgH92rINMGDXaWRnB9kd5WP5ViYJg)MHdLY)EYGEf71eMN3sgR)E)(vyoPRvW5LLXMgdh92FGt)vE9Qw8ZuYNC)x)rihmMQ(lasdFQglUnH8ZL9EBBrJ)uNwj1VuFWu1BJNExSN(OU76h60DQzTudYqNOVx6vLU4xoRx6LTS76hIdxUYGnm8FbwoAy4f6)(d
```

**Comment by emptyrivers** [comment]
I imported that group @mrbuds and could not figure out how to reproduce the issue. Can you provide steps on how to reproduce the issue you see?

**Comment by mrbuds** [comment]
rogue icon don't want to stay with his friends
prev
```
!nBvZYjooq4NM5WEyRAmdKAVgiMjufJHfz2FUOGWsy0gzjxsYHah4zF7wYgimKzYEBRIYi3sQ)97RBttO5uIJsEq8cLurjC8hiQ0A2rjJMTmlpDXOjlgnnLswBSCHDiR4zU1utjdvYdhyw(XvZz2ITvcTNsED2MnoHNM9zkz)fRz6ITg7CJep0OuuVDACKrzSGvZsWtM1l88lHN9rHWfa3HuZke4RK1GhaoyJMFPp8vM1kDg9XvdpT)Xv9alx4LgTlePEM1dlank1Y2vBGLUTH1WBERSSuyDr)j)KaCjxSUzZM891ck5X0PZhVCkSF4vwJLb2YvluQjCxuZUM1IxGScbUL8vk5Pr3tYFIKF)cqznOdqQvS9GYpD05wr4OK5PtbLhKbgtWu(TuIMvjCDEAJ(SR5JIkmQMk9TYut0NleYkgMrUmH0gfAJgEYHybpWtrHorHrdHeO)kMu)Xp9Pe7h78WnykzP(m8WA8rpnljPpLOmSa4SXjEQqXCo66qgVafw1O8YUutC3RL6Khexl0htsiGUdg67YrIG(lyQ2KzaSsjjaEx(Qq1jjLxkUuQvuc(CeLW3dLmzbwdQ7o)e9yRbQQdHiswbGGSbyAqT5kYHlwg7HyvVWQzQ)aWLTPdyBdu5aiPRqO5sDjw91q9u7hdiEisilU)Hjljq0ylMk0Laak7U768IoQzVZUv49(H870GJbEOe4yzIDyhbj)pLCujFbkTwajcudMNfYnr29yuAmWjJwKMMH8cpSXS6okyaKQ3ilJRBq9V5Ubdy9)Nd7jtI2jQcKCfDncw4YUlCvU8cvzzCzJdcIb)walhydaihTYjcCPc6LbvsV4vFYyiZhvxspuFgf3StNdBLQzRvcog0)0(wD3JStcEkCJIgN3uHQzznhroKM2)7oAaLaEHeIb8gqsgDSiOR176HEhLm2kpCC1V3W4y(94QCiiEwiQVhX6(fiLiGm)EiZnY2oVS459p0s6W7D2SXwUjHwUjHwUjNB5gsxT6F4S88zFBXKV(y(fE6yfReMDmBz(0jzW8HTcz5wec95aUK5rJgc6oM77WlVffgj5AO9xi58om3)p3kR4JLH7ncsKGRehEojJm5H0yjkscw349g9mipadkAlkFGIxGx9x7(7h)LbvJM)c)hpxZTfqekiKCiY8Td0(rJWUA607md7TZeppc7QjG3A46nMYfgN)Iipk8BgoqL)1KpFcXElG5LDKd8V3VFvOM01k4sAzSPrV(V9lCozYBZAdBtjFY9F9RqoymvNmaug(uDGCBW6Z19EBBrh(wNwi1pfFWu1BJ3ExSNE)oB9DD6o3SwQbyOtCQx6nHUHDUOx61TS76hggUCJ5AHW)fyz)E4l0)9d
```

File: WeakAurasOptions/RegionOptions/DynamicGroup.lua (hunk 3 above)
Comment (emptyrivers) [reply WeakAurasOptions/RegionOptions/DynamicGroup.lua]: No, I don't want to do it in this pull request.

**Comment by mrbuds** [comment]
> 
> 
> I imported that group @mrbuds and could not figure out how to reproduce the issue. Can you provide steps on how to reproduce the issue you see?

I have 5 buffs, one of them has no duration (weekly aura prev, limit=5, i add 2 buffs on myself and it kept showing the icon with wrong duration

prev
on this screen i have a 3days fade prev

It is easier to reproduce with grow=up, with grow=right i have to have one more buff on me to replicate

**Comment by emptyrivers** [comment]
@Stanzilla no, I don't really want to do that. Mostly because I have to do this at work and I hate it so much.

**Comment by mrbuds** [comment]
I confirm limit problems are fixed
I checked about the icon problem, it show with border and a preview with current master

File: WeakAuras/RegionTypes/DynamicGroup.lua [hunk 5]
```diff
@@ -673,8 +675,8 @@ local function modify(parent, region, data)
                       type(newPos[3]) ~= "boolean" and true or newPos[3]
       else
         x, y, show =  type(regionData.xOffset) == "number" and regionData.xOffset or 0,
-                        type(regionData.yOffset) == "number" and regionData.Offset or 0,
-                        type(regionData.shown) ~= "boolean" and true or regionData.shown
+                      type(regionData.yOffset) == "number" and regionData.yOffset or 0,
+                      false
```
Comment (InfusOnWoW) [inline WeakAuras/RegionTypes/DynamicGroup.lua]: That should be argueably a different commit
